### PR TITLE
Fix "Start at a given frame" in trace dialog 

### PR DESCRIPTION
**Pull request by pau-baiget**
Two commits:
* The fix.
* Adding some semantics to the startType selector for code clarity.

Fixes https://github.com/google/gapid/issues/3178

### CODE REVIEW

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 1]
```diff
@@ -257,6 +258,18 @@ protected void buttonPressed(int buttonId) {
...
+      private static final int TRACE_START_TYPE_FRAME = 2;
+      private static final HashMap<Integer, String> startTypeTitles = new HashMap<Integer, String>();
+
+      static
+      {
+        startTypeTitles.put(TRACE_START_TYPE_BEGINNING, "Beginning");
+        startTypeTitles.put(TRACE_START_TYPE_MANUAL, "Manual");
+        startTypeTitles.put(TRACE_START_TYPE_FRAME, "Frame");
+      };
+
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: Use an enum:
```
startType.setItems(Arrays.stream(StartType.values()).map(Enum::name).toArray(String[]::new));
...
startType.select(<enum>.ordinal());
...
startType.getSelectionIndex() == <enum>.ordinal()
...
etc.
...
private static enum StartType {
  Beginning, Manual, Frame;
}
```

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 2]
```diff
@@ -787,7 +787,7 @@ public TraceRequest getTraceRequest(Settings settings) {
           options.addAllEnvironment(splitEnv(envVars.getText()));
         }
         if (config.getMidExecutionCaptureSupport() != Service.FeatureStatus.NotSupported) {
-          options.setDeferStart(settings.traceStartAt != 0);
+          options.setDeferStart(settings.traceStartAt < 0);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: Yikes. We probably should also fix this in the server... starting a trace at frame x is by definition a deferred start.

Comment (pau-baiget) [reply gapic/src/main/com/google/gapid/views/TracerDialog.java]: Yes, it's a bit misleading. I think deferred start refers to "Start when the users clicks button".

File: gapic/src/main/com/google/gapid/views/TracerDialog.java (hunk 1 above)
Comment (pau-baiget) [reply gapic/src/main/com/google/gapid/views/TracerDialog.java]: Nifty! Will do that. Thanks!

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 3]
```diff
@@ -257,6 +259,19 @@ protected void buttonPressed(int buttonId) {
...
+
+      private static final HashMap<Integer, String> startTypeTitles = new HashMap<Integer, String>();
+
+      static
+      {
+        startTypeTitles.put(StartType.Beginning.ordinal(), "Beginning");
+        startTypeTitles.put(StartType.Manual.ordinal(), "Manual");
+        startTypeTitles.put(StartType.Frame.ordinal(), "Frame");
+      };
+
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: Just use `startType.name()`. If, however, you don't want to tie yourself to the name of the enum, add a final field in the enum for it. Something like:
```
private static enum StartType {
  Beginning("Beginning"), ... ;

  public final String display;

  private StartType(...) { ... }
}
```

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 4]
```diff
@@ -294,6 +309,8 @@ protected void buttonPressed(int buttonId) {
       protected boolean userHasChangedOutputFile = false;
       protected boolean userHasChangedTarget = false;
 
+
+
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: remove

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 5]
```diff
@@ -367,19 +384,19 @@ protected String createAndShowDialog(String current) {
             new GridData(GridData.FILL_HORIZONTAL));
         createLabel(durGroup, "Start at:");
         startType = Widgets.createDropDown(durGroup);
-        startType.setItems("Beginning", "Manual", "Frame");
+        startType.setItems(Arrays.stream(StartType.values()).map(Enum::name).toArray(String[]::new));
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: This isn't using the map you created above, so if you choose to not use the name, then this needs to be fixed to whatever you choose.

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 6]
```diff
@@ -457,16 +474,16 @@ protected void configureDialog(DirectoryDialog dialog) {
 
         Listener mecListener = e -> {
           TraceTypeCapabilities config = getSelectedApi();
-          boolean beginning = startType.getSelectionIndex() == 0;
-          if (!beginning && config != null &&
+          boolean beginning = startType.getSelectionIndex() == StartType.Beginning.ordinal();
+          if (beginning && config != null &&
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: oops, this inverts the test!

**Pull request merged**